packages/styled-base/types/helper.d.ts

```diff
@@ -5,7 +5,7 @@ import * as React from 'react'
  */
 export type PropsOf<
```

Comment:
how this type is different from "builtin" `ComponentProps`?
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/6d7624e50fcad42e22a1cfddcea4bc7fa03bc29d/types/react/index.d.ts#L756-L760
Also - what about that suggestion there that we should prefer using `ComponentPropsWithRef` or `ComponentPropsWithoutRef` appropriately over `ComponentProps`?

Reply:
ComponentProps appears it doesn't take into account default props.

Using ComponentPropsWithRef instead of ComponentProps causes this test to fail:
`const AdditionallyStyledDiv = themedStyled(StyledDiv)({})`

Finally, without changing ComponentPropsWithRef to just ComponentProps `const StyledClass0 = styled(ReactClassComponent0)({})` has a ref property which is not correct.

As far as I can see any changes around these props will cause some tests to fail.

Reply:
> ComponentProps appears it doesn't take into account default props.

Could you elaborate more on that?

Reply:
If a class component has `defaultProps` on it, PropsOf will end up with those props being optional because they were provided by default props